ini/parse.ts

```diff
@@ -1,9 +1,64 @@
 // Copyright 2018-2025 the Deno authors. MIT license.
 // This module is browser compatible.
 
-import { IniMap, type ReviverFunction } from "./_ini_map.ts";
+import type { ReviverFunction } from "./_ini_map.ts";
 export type { ReviverFunction };
 
+const SECTION_REGEXP = /^\[(?<name>.+)]$/;
+const KEY_VALUE_REGEXP = /^\s*(?<key>.*?)\s*=\s*(?<value>.*?)\s*$/;
+
+function trimQuotes(value: string): string {
+  if (value.startsWith('"') && value.endsWith('"')) {
+    return value.slice(1, -1);
+  }
+  return value;
+}
+
+/** Detect supported comment styles. */
+function isComment(input: string): boolean {
+  return (
+    input.startsWith("#") ||
+    input.startsWith(";") ||
+    input.startsWith("//")
+  );
+}
+
+/** Detect a section start. */
+function isSection(input: string, lineNumber: number): boolean {
+  if (input.startsWith("[")) {
+    if (input.endsWith("]")) {
+      return true;
+    }
+    throw new SyntaxError(
+      `Unexpected end of INI section at line ${lineNumber}`,
+    );
+  }
+  return false;
+}
+
+function* readTextLines(text: string): Generator<string> {
+  let line = "";
+  for (let i = 0; i < text.length; i += 1) {
+    const char = text[i];
+    switch (char) {
+      case "\n":
+        yield line;
+        line = "";
+        break;
+      case "\r":
+        yield line;
+        line = "";
+        if (text[i + 1] === "\n") i += 1;
+        break;
+      default:
+        if (line === " " || line === "\t") continue;
```

Comment:
What does this condition try to do?

Now `parse(" value=true")` throws, but it returns `{ value: true }` on main.